File: 백준/Silver/7568. 덩치/덩치.py
n = int(input())
arr = []
for i in range(n):
  w,h = map(int,input().split())
  arr.append([w,h])


answer = [0 for i in range(n)]
for i in range(n):
  for j in range(n):
    if arr[i][0] < arr[j][0] and arr[i][1] < arr[j][1]:
      answer[i] +=1

for i in range(n):
  print(answer[i]+1 ,end=' ')

File: 백준/Silver/7568. 덩치/덩치.js
const input = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n")

const N = Number(input[0])

const people = input.slice(1).map((row) => row.split(" ").map(Number))
const answer = []
for (let i = 0; i < N; i += 1) {
  const [ta, tb] = people[i]
  let count = 0
  for (let j = 0; j < N; j += 1) {
    if (i === j) continue
    const [oa, ob] = people[j]
    if (ta < oa && tb < ob) {
      count += 1
    }
  }
  answer.push(count+1)
}
console.log(answer.join(' '))
